src/domain/opendatafabric/src/identity/dataset_identity.rs

```diff
@@ -104,18 +106,18 @@ macro_rules! impl_serde {
 }
 
 pub(crate) use impl_serde;
-use like::Like;
+use like::ILike;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-macro_rules! newtype_str {
+macro_rules! newtype_istr {
     ($typ:ident, $parse:expr, $visitor:ident) => {
         #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
         pub struct $typ(Arc<str>);
 
         impl $typ {
             pub fn new_unchecked<S: AsRef<str> + ?Sized>(s: &S) -> Self {
-                Self(Arc::from(s.as_ref()))
+                Self(Arc::from(Self::into_lowercase(s.as_ref())))
```

Comment:
There is a big difference between case-insensitive comparison and forcing all values to lowercase. If user wants to have a dataset called `myFancyDataset` - I think we should display it under this **exact** name, but it should be considered the same as `myfancydataset`.

> **Aside:** The first item on the ticket is to `Spell out how case (in)sensitivity works in ODF spec` - i.e. do a design in the form of a PR for ODF spec. This is where I hoped we would discuss such things - not during implementation.

I think the approach I propose is closer to how DNS works and how GitHub account names work.

CC @zaychenko-sergei what do you think?

Reply:
My approach will of course would complicate `Eq, Ord, Hash` implementations, making them do conversions each time. But it shouldn't be a big deal for `Hash` and we don't do that many ordering comparisons to get a performance hit.

Reply:
I don't see a good use case where dataset would need a "display name" that is different from its name (symbolic ID) in any way other than the character case.

Accounts in github do have separate `login` (e.g. `sergiimk`) and `name` (e.g. `Sergii M`), but you only see the latter in user info - never when addressing a user. I wouldn't want us to have an account name, account display name, and then also account full name.

In general I was hoping to avoid enforcing any additional naming restrictions on account/dataset names other than making sure that they are valid DNS substrings and behave like those.

We could force to lowercase and push back the decision to support upper case names, but then we'll need to accept that users logging in via external OAuth providers like GH will have their account names lowercased, which is not ideal.